src/lib/Makefile

```diff
@@ -527,7 +527,7 @@ else
 	@echo "PLUMED_RUNTIME_LOAD=" $(realpath $(OBJ_DYNAMIC_WRAPPER)) $(LIBS) $(LDFLAGS) > $@
 endif
 	@echo "PLUMED_STATIC_LOAD=" $(realpath $(OBJ_KERNEL)) $(realpath $(OBJ_WRAPPER)) $(LIBS) $(DYNAMIC_LIBS) $(LDFLAGS) >> $@
-	@echo "PLUMED_SHARED_LOAD=" $(realpath .)/libplumed.$(SOEXT) $(LIBS) $(LDFLAGS) >> $@
+	@echo "PLUMED_SHARED_LOAD=" $(realpath .)/libplumed.$(SOEXT) $(LIBS) -Wl,-rpath-link $(realpath .) $(LDFLAGS) >> $@
```

Comment:
On conda packaging we are adding this flag by hand in the `STATIC_LIBS` variable (that then gets inserted in the `LIBS` make variable - this is for historical reasons). Did you try passing it in that way? I don't think it is generally portable (what about MacOS?) nor needed (if you use env modules to load plumed, which set `LD_LIBRARY_PATH`, it is not needed), so I would not include it here

Reply:
No, I didn't; actually, I'm not sure where to find the `STATIC_LIBS` variable -- the error occurs during the `make test` stage:

```[  882s] g++ -g -O2 -fdebug-prefix-map=/usr/src/packages/BUILD=. -fstack-protector-strong -Wformat -Werror=format-security -fPIC -Wall -pedantic -std=c++11 main.o -o plumed /usr/src/packages/BUILD/src/lib/libplumed.so -ldl
[  882s] /usr/bin/ld: warning: libplumedKernel.so, needed by /usr/src/packages/BUILD/src/lib/libplumed.so, not found (try using -rpath or -rpath-link)
[  882s] /usr/bin/ld: /usr/src/packages/BUILD/src/lib/libplumed.so: undefined reference to 'plumed_symbol_table_init'
[  882s] /usr/bin/ld: /usr/src/packages/BUILD/src/lib/libplumed.so: undefined reference to 'plumed_symbol_table'
[  882s] collect2: error: ld returned 1 exit status
[  882s] make[3]: *** [Makefile:18: plumed] Error 1
[  882s] make[3]: Leaving directory '/usr/src/packages/BUILD/test/link'
```

Reply:
It's [here](https://github.com/plumed/plumed2/blob/23f493cd559db4c13b7d03dc15f5f81130ab4448/conda/plumed/build.sh#L8). Before building, just add
```
  export STATIC_LIBS=-Wl,-rpath-link,$PREFIX/lib
```

I would recommend also adding `-ltr` to `LIBS` (as it is done [here](https://github.com/plumed/plumed2/blob/23f493cd559db4c13b7d03dc15f5f81130ab4448/conda/plumed/build.sh#L9) when you build newer PLUMED versions, since this will allow enabling internal asmjit that makes calculations faster.


Notice that the `STATIC_LIBS` is a non standard env var. I added it to keep separate (a) variables needed when linking the plumed library (`LIBS`) and (b) variables needed when linking programs depending on the plumed library (`STATIC_LIBS`). On MacOS, the latter is empty since libraries by default store the absolute path of other libraries they depend on. Also on Linux it can be empty when using env modules (that set `LD_LIBRARY_PATH`). However, Linux systems using rpath need this to be set.